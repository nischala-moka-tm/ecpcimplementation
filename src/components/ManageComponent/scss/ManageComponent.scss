@import "../../../shared/scss/Common-Style.scss";

.manage-component {
  .sub-head {
    border-bottom: 1px solid #000;

    p {
      background: #191717 0% 0% no-repeat padding-box;
      border: 1px solid #ccc5c5;
      width: 260px;
      color: #fff;
      padding: 10px 20px;
      margin-bottom: -1px;
      text-align: center;
      border-bottom: 0;
    }
  }

  .manage-body {
    min-height: 300px;
    margin: 20px 0 10px 20px;
    position: relative;

    @media (min-width: $breakpoint-tablet) {
      margin: 20px 0 10px 35px;
    }

    .reference-guide,
    .info-text {
      font-size: 12px;
      line-height: 14px;
      margin: 10px 0;
      padding-right: 20px;
    }

    .status-codes {
      font-size: 12px;
      line-height: 18px;
      padding-left: 15px;

      @media (min-width: $breakpoint-tablet) {
        position: absolute;
        top: 0;
        right: 25px;
      }

      ul {
        padding-left: 0;

        li {
          list-style: none;
          position: relative;

          .status-roundend {
            height: 6px;
            width: 6px;
            display: inline-block;
            position: static;
            border-radius: 50%;
            margin: 2px 5px;
          }
        }
      }
    }

    .categories-list {
      position: relative;
      display: block;

      div {
        &.heading {
          margin-bottom: 5px;

          .heading-text {
            color: #1d1d1d;
            font-weight: bold;

            .status-roundend {
              height: 8px;
              width: 8px;
              display: inline-block;
              position: static;
              border-radius: 50%;
              margin-right: 3px;
            }
          }

          button {
            &.plusmenu-danger {
              cursor: pointer;
              border: none;
              border-radius: 50%;
              background: transparent;

              i {
                position: relative;

                &.minus-checked {
                  &::after {
                    content: "";
                    background: url("../../../assets/Group 162.svg");
                    width: 20px;
                    height: 22px;
                    position: absolute;
                    top: -18px;
                    left: -15px;
                    background-repeat: no-repeat;
                  }
                }

                &.plus-checked {
                  &::after {
                    content: "";
                    background: url("../../../assets/Group 410.svg");
                    width: 30px;
                    height: 30px;
                    position: absolute;
                    top: -18px;
                    left: -15px;
                    background-repeat: no-repeat;
                  }
                }
              }
            }
          }

          &:hover {
            .dropdown-content {
              display: inline-block;
              transition: 0.3s;
            }
          }

          .dropdown-content {
            display: none;
            position: absolute;
            width: 100%;
            z-index: 100;

            .selectble-popup {
              padding: 0;
              background-color: #fff;
              box-sizing: border-box;
              border: 1px solid #707070;
              max-width: fit-content;
              display: flex;
              flex-direction: column;

              li {
                padding: 0.2em 1.25em;
                list-style: none;
                font-family: $toyota-book;
                font-size: 12px;
                justify-content: start;
                align-items: center;
                text-align: left;

                &:hover {
                  background-color: #000;
                  color: #fff;
                  width: auto;
                  cursor: pointer;
                }
              }
            }
          }
        }
      }

      .sub-category {
        position: relative;
        display: block;

        ul {
          padding-left: 30px;
          margin-bottom: 5px;

          li {
            list-style: none;
            font-size: 14px;
            line-height: 20px;

            .status-roundend {
              height: 8px;
              width: 8px;
              display: inline-block;
              position: static;
              border-radius: 50%;
              margin-right: 3px;
            }

            button {
              &.plusmenu-danger {
                cursor: pointer;
                border: none;
                border-radius: 50%;
                background: transparent;

                i {
                  position: relative;

                  &.minus-checked {
                    background: transparent;

                    &::after {
                      content: "";
                      background: url("../../../assets/Group 162.svg");
                      width: 20px;
                      height: 22px;
                      position: absolute;
                      top: -18px;
                      left: -15px;
                      background-repeat: no-repeat;
                    }
                  }

                  &.plus-checked {
                    &::after {
                      content: "";
                      background: url("../../../assets/Group 410.svg");
                      width: 30px;
                      height: 30px;
                      position: absolute;
                      top: -18px;
                      left: -15px;
                      background-repeat: no-repeat;
                    }
                  }
                }
              }
            }

            &:hover {
              .dropdown-content {
                display: inline-block;
              }
            }

            .dropdown-content {
              display: none;
              position: absolute;
              width: 100%;
              z-index: 100;

              .selectble-popup {
                padding: 0;
                background-color: #fff;
                box-sizing: border-box;
                border: 1px solid #707070;
                max-width: fit-content;
                display: flex;
                flex-direction: column;

                li {
                  padding: 0.2em 1.25em;
                  list-style: none;
                  font-family: $toyota-book;
                  font-size: 12px;
                  justify-content: start;
                  align-items: center;
                  text-align: left;

                  &:hover {
                    background-color: #000;
                    color: #fff;
                    width: auto;
                    cursor: pointer;
                  }
                }
              }
            }
          }

          p {
            &.category-item {
              padding-left: 50px;
            }
          }
        }

        .sub-category {
          ul {
            li {
              font-size: 12px;
              line-height: 18px;
            }
          }
        }

        .final-category {
          ul {
            padding-left: 38px;
            margin-bottom: 0;

            li {
              font-size: 12px;
              line-height: 18px;
            }
          }
        }
      }
    }

    .btn {
      width: 200px;
      height: 35px;
      font-size: 16px;
      line-height: 16px;

      &:focus {
        box-shadow: none;
      }

      &.btn-primary {
        background: #fff;
        color: #000;
        border: 1px solid #707070;
        border-radius: 1px;
        margin-right: 20px;
      }
    }
  }

  .button-options {
    text-align: right;
    margin: 15px 20px 35px 0;

    @media (min-width: $breakpoint-tablet) {
      margin: 15px 60px 35px 0;
    }

    .btn {
      height: 55px;
      font-size: 16px;
      line-height: 16px;
      padding: 0 20px;

      &:focus {
        box-shadow: none;
      }

      &.btn-primary {
        background: #fff;
        color: #000;
        border: 1px solid #707070;
        border-radius: 1px;
        margin-right: 20px;
      }

      &.btn-secondary {
        color: #fff;
        background: #000000 0% 0% no-repeat padding-box;
        border-radius: 1px;
      }
    }
  }
}

// Submit for Approval Pop Up

.approval-popup {
  .modal-dialog {
    max-width: 450px;
    position: absolute;
    top: 20%;

    @media (min-width: $breakpoint-tablet) {
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%) !important;
    }

    .modal-content {
      padding-bottom: 0;
    }

    .modal-header {
      border-bottom: none;
    }

    .modal-body {
      padding: 1.5rem;
      text-align: center;

      @media (min-width: $breakpoint-tablet) {
        padding: 1.5rem 2.5rem;
      }

      p {
        font-family: $toyota-book;
        font-size: 20px;
        line-height: 36px;
        color: #000;
      }

      .button-options {
        justify-content: center;

        .btn {
          &.btn-primary {
            margin: 10px 0;
            width: 200px;
          }
        }
      }
    }
  }
}