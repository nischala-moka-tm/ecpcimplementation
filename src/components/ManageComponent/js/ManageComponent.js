import React, { useState, useEffect } from "react";
import { Button, Modal } from "react-bootstrap";
import "../scss/ManageComponent.scss";
import statusDetails from "../../CategoriesData/StatusDetails.json";
import AddNewCategory from "../../AddNewCategory/js/AddNewCategory";
import AddPreferenceLevels from "../../AddPreferenceLevels/js/AddPreferenceLevels";
import DetailedViewPage from "../../DetailedViewPage/js/DetailedViewPage";
import { AxiosGet } from "../../AxiosMethods/ApiCalls";
import AddPermissionLevels from "../../AddPermissionLevels/js/AddPermissionLevels";

import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { FaPlusCircle, FaMinusCircle } from "react-icons/fa";

const SubCategory = (props) => {
	const [subCategory, setSubCategory] = useState(false);
	return (
		<div className="sub-category">
			<ul>
				<li>
					<button
						className="plusmenu-danger"
						onClick={() => setSubCategory(!subCategory)}
					>
						{props.subCategory &&
							(subCategory ? <FaMinusCircle /> : <FaPlusCircle />)}
					</button>
					<Status status={props.status} />
					{props.subCategoryName}
					<SelectablePopupLevel {...props} />
				</li>

				{props.subCategory &&
					props.subCategory.map((category, index) => {
						return (
							subCategory && (
								<SubCategory key={index} {...category} notify={props.notify} />
							)
						);
					})}
			</ul>
		</div>
	);
};

function SelectablePopupLevel(props) {
	const [showLevelPopup, setLevelPopup] = useState(false);
	const [optionType, setOptionType] = useState("Add");
	const [showDetailLevel, setShowDetailLevel] = useState(false);
	const LevelCondition1 = (category, optType) => {
		return (
			<AddPermissionLevels
				show={showLevelPopup}
				onClose={() => setLevelPopup(false)}
				category={category}
				optionType={optType}
				brand={category.brand}
				notify={props.notify}
				level={optType === "Add" ? category.level + 1 : category.level}
			/>
		);
	};
	const LevelCondition2 = (category, optType) => {
		return (
			<AddPreferenceLevels
				show={showLevelPopup}
				onClose={() => setLevelPopup(false)}
				category={category}
				optionType={optType}
				brand={category.brand}
				notify={props.notify}
				level={optType === "Add" ? category.level + 1 : category.level}
			/>
		);
	};

	return (
		<div className="dropdown-content">
			<ul className="selectble-popup">
				<li
					onClick={() => {
						setLevelPopup(true);
						setOptionType("Add");
					}}
				>
					Add New
					{props.id.includes("PREFERENCES") ? " Preference" : " Permission"}
				</li>
				<li
					onClick={() => {
						setLevelPopup(true);
						setOptionType("Edit");
					}}
				>
					Modify
				</li>
				<li onClick={() => setShowDetailLevel(true)}>View Existing Content</li>
				<li
					onClick={() => {
						setLevelPopup(true);
						setOptionType("Delete");
					}}
				>
					Delete
				</li>
			</ul>
			{showLevelPopup &&
				(props.id.includes("PREFERENCES")
					? LevelCondition2(props, optionType)
					: LevelCondition1(props, optionType))}
			{showDetailLevel && (
				<DetailedViewPage
					show={setShowDetailLevel}
					onClose={() => setShowDetailLevel(false)}
				/>
			)}
		</div>
	);
}
const SubmitApproval = (props) => {
	return (
		<Modal
			show={props.SubmitPref}
			onHide={props.handleClose}
			className="approval-popup modalpopup"
		>
			<Modal.Header closeButton></Modal.Header>
			<Modal.Body>
				<p>Your new permissions category has been submitted for approval</p>
				<div className="button-options">
					<Button variant="primary" size="sm" onClick={props.handleClose}>
						OK
					</Button>
				</div>
			</Modal.Body>
		</Modal>
	);
};
function MainCategory({ category, brand, notify }) {
	const [SubC1, setSubC1] = useState(false);

	return (
		<div className="categories-list">
			<div className="heading">
				<button className="plusmenu-danger" onClick={() => setSubC1(!SubC1)}>
					{category.subCategory &&
						(SubC1 ? <FaMinusCircle /> : <FaPlusCircle />)}
				</button>
				<span className="heading-text">
					<Status status={category.status} />
					{category.categoryName}
				</span>
				<SelectablePopupLevel {...category} brand={brand} notify={notify} />
			</div>

			{category.subCategory &&
				category.subCategory.map((subCateogry, pos) => {
					return (
						SubC1 && (
							<SubCategory
								key={pos}
								{...subCateogry}
								brand={brand}
								notify={notify}
							/>
						)
					);
				})}
		</div>
	);
}
function ManageComponent(props) {
	const [showLevel1, SetLevel1] = useState(false);
	const [SubmitPref, setSubmitPref] = useState(false);
	const [isLoading, setLoading] = useState(true);
	const [categorydata, setCategoriesData] = useState([]);
	const handleClose = () => setSubmitPref(false);

	useEffect(() => {
		getApiCall();
	}, [props.brand, props.type]);
	const getApiCall = () => {
		setLoading(true);
		const getDataApi = AxiosGet({
			brand: props.brand,
			type: props.type,
		});
		getDataApi.then((result) => {
			setLoading(false);
			setCategoriesData(result.data.data);
		});
	};
	const notify = (res, resType) => {
		if (resType === "success") {
			toast.success(res);
			getApiCall();
		} else {
			toast.error(res);
		}
	};

	return (
		<div className="manage-component" id="manage-permission">
			<div className="sub-head">
				<p>Permissions/Preferences</p>
			</div>
			<div className="manage-body">
				<ToastContainer />
				<div className="status-codes">
					<ul>
						{statusDetails.map((status, index) => {
							return (
								<li key={index}>
									<span
										className="status-roundend"
										style={{ backgroundColor: status.colorCode }}
									></span>
									{status.status}
								</li>
							);
						})}
					</ul>
				</div>
				<p className="reference-guide">
					<a href="/">Click Here</a> for reference guide
				</p>
				<Button variant="primary" size="sm" onClick={() => SetLevel1(true)}>
					Create new Category
				</Button>
				<p className="info-text">
					To edit/delete existing category or subcategory right click on item
				</p>

				{isLoading
					? "Loading...."
					: categorydata &&
					  categorydata.map((category, index) => {
							return (
								<MainCategory
									category={category}
									key={index}
									brand={props.brand}
									notify={notify}
								/>
							);
					  })}
			</div>

			<div className="button-options">
				<Button
					variant="secondary"
					size="sm"
					onClick={() => setSubmitPref(!SubmitPref)}
				>
					Submit for Approval
				</Button>
				{SubmitPref && (
					<SubmitApproval SubmitPref={SubmitPref} handleClose={handleClose} />
				)}
			</div>
			{showLevel1 && (
				<AddNewCategory show={showLevel1} onClose={() => SetLevel1(false)} />
			)}
		</div>
	);
}
function Status(props) {
	let bgColor = "";
	statusDetails.map((status) => {
		if (status.status === props.status) {
			bgColor = status.colorCode;
		}
	});
	return (
		<span
			className="status-roundend"
			style={{ backgroundColor: bgColor }}
		></span>
	);
}

export default ManageComponent;
