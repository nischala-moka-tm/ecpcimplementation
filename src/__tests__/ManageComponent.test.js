import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import ManageComponent from "../components/ManageComponent/js/ManageComponent";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
Enzyme.configure({ adapter: new Adapter() });

const brand = "Toyota";
const type = "dashboard";

it("Searching rendering id correctly", async () => {
  const wrapper = mount(<ManageComponent brand={brand} type={type} />);
  expect(wrapper.find("manage-permission")).toBeDefined();
  expect(wrapper.find("status-codes")).toBeDefined();
});

it("renders correctly when there are no items", () => {
  const tree = renderer
    .create(<ManageComponent brand={brand} type={type} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("renders <DashboardComponent />", () => {
  it("should pass mount snapshot test", async () => {
    let wrapper;
    await act(async () => {
      wrapper = mount(<ManageComponent brand={brand} type={type} />);
    });
    expect(wrapper).toMatchSnapshot();
  });
});
