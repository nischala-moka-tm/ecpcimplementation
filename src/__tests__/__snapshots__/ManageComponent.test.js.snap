// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <DashboardComponent /> should pass mount snapshot test 1`] = `ReactWrapper {}`;

exports[`renders correctly when there are no items 1`] = `
<div
  className="manage-component"
  id="manage-permission"
>
  <div
    className="sub-head"
  >
    <p>
      Permissions/Preferences
    </p>
  </div>
  <div
    className="manage-body"
  >
    <div
      className="Toastify"
    />
    <div
      className="status-codes"
    >
      <ul>
        <li>
          <span
            className="status-roundend"
            style={
              Object {
                "backgroundColor": "#E86D0F",
              }
            }
          />
          Pending Approval
        </li>
        <li>
          <span
            className="status-roundend"
            style={
              Object {
                "backgroundColor": "#00FF00",
              }
            }
          />
          Live
        </li>
        <li>
          <span
            className="status-roundend"
            style={
              Object {
                "backgroundColor": "#0346FF",
              }
            }
          />
          Approved Pending Go-Live
        </li>
        <li>
          <span
            className="status-roundend"
            style={
              Object {
                "backgroundColor": "#FFEE03",
              }
            }
          />
          Approved Expiring Soon
        </li>
        <li>
          <span
            className="status-roundend"
            style={
              Object {
                "backgroundColor": "#CECECE",
              }
            }
          />
          Inactive
        </li>
        <li>
          <span
            className="status-roundend"
            style={
              Object {
                "backgroundColor": "#0099ff",
              }
            }
          />
          Draft
        </li>
      </ul>
    </div>
    <p
      className="reference-guide"
    >
      <a
        href="/"
      >
        Click Here
      </a>
       for reference guide
    </p>
    <button
      className="btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Create new Category
    </button>
    <p
      className="info-text"
    >
      To edit/delete existing category or subcategory right click on item
    </p>
    Loading....
  </div>
  <div
    className="button-options"
  >
    <button
      className="btn btn-secondary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Submit for Approval
    </button>
  </div>
</div>
`;
